ush/setup.py

```diff
@@ -1484,14 +1509,66 @@ def _dict_find(user_dict, substring):
-        logging.debug(f'New fix file mapping:\n{fixed_files["CYCLEDIR_LINKS_TO_FIXam_FILES_MAPPING"]=}')
+        logger.debug(f'New fix file list:\n{fixed_files["FIXgsm_FILES_TO_COPY_TO_FIXam"]=}')
+        logger.debug(f'New fix file mapping:\n{fixed_files["CYCLEDIR_LINKS_TO_FIXam_FILES_MAPPING"]=}')
+
 
+    # -----------------------------------------------------------------------
+    #
+    # Check that UFS FIRE settings are correct and consistent
+    #
+    # -----------------------------------------------------------------------
+    fire_conf = expt_config["fire"]
+    if fire_conf["UFS_FIRE"]:
+        if build_config["Application"]!="ATMF":
+            raise Exception("UFS_FIRE == True but UFS SRW has not been built for fire coupling; see users guide for details")
+        fire_input_file=os.path.join(fire_conf["FIRE_INPUT_DIR"],"geo_em.d01.nc")
+        if not os.path.isfile(fire_input_file):
+            raise FileNotFoundError(
+                dedent(
+                    f"""
+                The fire input file (geo_em.d01.nc) does not exist in the specified directory:
+                {fire_conf["FIRE_INPUT_DIR"]}
+                Check that the specified path is correct, and that the file exists and is readable
+                """
+                )
+            )
+        if fire_conf["FIRE_NUM_TASKS"] < 1:
+            raise ValueError("FIRE_NUM_TASKS must be > 0 if UFS_FIRE is True")
+        elif fire_conf["FIRE_NUM_TASKS"] > 1:
+            raise ValueError("FIRE_NUM_TASKS > 1 not yet supported")
+
+        if fire_conf["FIRE_NUM_IGNITIONS"] > 5:
+            raise ValueError(f"Only 5 or fewer fire ignitions supported")
+
+        if fire_conf["FIRE_NUM_IGNITIONS"] > 1:
+            # These settings all need to be lists for multiple fire ignitions
+            each_fire = ["FIRE_IGNITION_ROS", "FIRE_IGNITION_START_LAT", "FIRE_IGNITION_START_LON",
+                         "FIRE_IGNITION_END_LAT", "FIRE_IGNITION_END_LON", "FIRE_IGNITION_RADIUS",
+                         "FIRE_IGNITION_START_TIME", "FIRE_IGNITION_END_TIME"]
+            for setting in each_fire:
+                if not isinstance(fire_conf[setting], list):
+                    logger.critical(f"{fire_conf['FIRE_NUM_IGNITIONS']=}")
+                    logger.critical(f"{fire_conf[setting]=}")
+                    raise ValueError(f"For FIRE_NUM_IGNITIONS > 1, {setting} must be a list of the same length")
+                if len(fire_conf[setting]) != fire_conf["FIRE_NUM_IGNITIONS"]:
+                    logger.critical(f"{fire_conf['FIRE_NUM_IGNITIONS']=}")
+                    logger.critical(f"{fire_conf[setting]=}")
+                    raise ValueError(f"For FIRE_NUM_IGNITIONS > 1, {setting} must be a list of the same length")
+
+        if fire_conf["FIRE_ATM_FEEDBACK"] > 0.0:
+            raise ValueError("FIRE_ATM_FEEDBACK > 0 (two-way coupling) not supported in UFS yet")
```

Comment:
The new WE2E tests and the `config.fire.yaml` sample configuration set `FIRE_ATM_FEEDBACK` to `1.0`.  This is causing the tests to fail since it is larger than `0.0`.  Should the test configurations be updated to use `FIRE_ATM_FEEDBACK: 0.0`, rather than `1.0`?

Reply:
Sorry about that, I updated the settings in the WE2E tests but failed to update it here. This is an issue that we found after merging the capability into the weather model where the two-way coupling was not actually working (results were identical to 1-way coupling); this just ensures that users don't try to use two-way coupling while it's not working in the initial version.

Reply:
That makes sense to me!  This has been resolved as of [e67173f](https://github.com/ufs-community/ufs-srweather-app/pull/1139/commits/e67173fb206c3d38bf78318b8eebfa5c4d0be035).